src/vmm/src/device_manager/mmio.rs

```diff
@@ -249,8 +249,9 @@ impl MMIODeviceManager {
         &mut self,
         vm: &VmFd,
         serial: Arc<Mutex<devices::legacy::Serial>>,
+        dev_info_opt: Option<MMIODeviceInfo>,
```

Comment:
nit: This works, but pls change function doc-comment to capture this change.

------------

For virtio devices the approach was to use two separate functions:
- `register_virtio_mmio_device()` does the actual registering
- `register_new_virtio_mmio_device()` allocates slot and calls the above. Also populates cmdline

You could do the same for Serial and RTC if you think it looks better.